pkg/piecestore/rpc/server/psdb/psdb.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package psdb
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+
+	_ "github.com/mattn/go-sqlite3" // sqlite is weird and needs underscore
+
+	"storj.io/storj/pkg/piecestore"
+	pb "storj.io/storj/protos/piecestore"
+)
+
+// PSDB -- Piecestore database
+type PSDB struct {
+	DB *sql.DB
+}
+
+// NewPSDB -- creates
+func NewPSDB(DBPath string) (*PSDB, error) {
+	if err := os.MkdirAll(filepath.Dir(DBPath), 0700); err != nil {
+		return nil, err
+	}
+
+	db, err := sql.Open("sqlite3", DBPath)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = db.Exec("CREATE TABLE IF NOT EXISTS `ttl` (`id` TEXT UNIQUE, `created` INT(10), `expires` INT(10));")
+	if err != nil {
+		return nil, err
+	}
+	_, err = db.Exec("CREATE TABLE IF NOT EXISTS `bandwidth_agreements` (`agreement` BLOB, `signature` BLOB);")
+	if err != nil {
+		return nil, err
+	}
+
+	return &PSDB{db}, nil
+}
+
+// checkEntries -- checks for and deletes expired TTL entries
+func checkEntries(dir string, rows *sql.Rows) error {
+
+	for rows.Next() {
+		var expID string
+
+		err := rows.Scan(&expID)
+		if err != nil {
+			return err
+		}
+
+		// delete file on local machine
+		err = pstore.Delete(expID, dir)
```

Comment:
To me this looks like it just deletes every row passed in. I see you're calling it from `DeleteExpiredEntries`, which contains the actual logic for finding the expired TTL entries. Because the logic to find/check those entries is not in this function, the name `checkEntries` doesn't make sense to me. I think `deleteRows` is a better  name for this function.